seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/checkpoint/CheckpointType.java

```diff
@@ -31,6 +31,15 @@ public enum CheckpointType {
     private final boolean auto;
     private final String name;
 
+    public static CheckpointType fromName(String name) {
+        for (CheckpointType type : CheckpointType.values()) {
+            if (type.name.equals(name)) {
+                return type;
+            }
+        }
+        throw new IllegalArgumentException("Unknown checkpoint type: " + name);
```

Comment:
CheckpointException is better?

Reply:
There just value convert to enum. I think we don't need `CheckpointException`.

Reply:
OK